File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Color Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var c = new Tinkerforge.BrickletColorV2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Set period for color callback to 0.1s (100ms)
        c.setColorCallbackConfiguration(100, false);
    }
);

// Register color callback
c.on(Tinkerforge.BrickletColorV2.CALLBACK_COLOR,
    // Callback function for color callback
    function (r, g, b, c) {
        console.log('Color [R]: ' + r);
        console.log('Color [G]: ' + g);
        console.log('Color [B]: ' + b);
        console.log('Color [C]: ' + c);
        console.log();
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_color_v2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Color Bricklet 2.0

ipcon = IPConnection.new # Create IP connection
c = BrickletColorV2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register color callback
c.register_callback(BrickletColorV2::CALLBACK_COLOR) do |r, g, b, c_|
  puts "Color [R]: #{r}"
  puts "Color [G]: #{g}"
  puts "Color [B]: #{b}"
  puts "Color [C]: #{c_}"
  puts ''
end

# Set period for color callback to 0.1s (100ms)
c.set_color_callback_configuration 100, false

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
